src/consent-logs.ts

```diff
@@ -0,0 +1,84 @@
+import * as Cookies from 'js-cookie';
+import { ConsentString, VendorList } from 'consent-string';
+import {
+    IAB_CMP_ID,
+    IAB_CMP_VERSION,
+    IAB_CONSENT_SCREEN,
+    IAB_CONSENT_LANGUAGE,
+} from './config';
+import { writeIabCookie } from './cookies';
+import { updateStateOnSave } from './cmp';
+import { IabPurposeState } from './types';
+
+const DUMMY_BROWSER_ID = `No bwid available`;
+
+type MsgData = {
+    iabVendorList: VendorList;
+    allowedPurposes: number[];
+    allowedVendors: number[];
+    isProd: boolean;
+};
+
+export const logConsent = ({
+    iabVendorList,
+    allowedPurposes,
+    allowedVendors,
+    isProd,
+}: MsgData): Promise<Response> => {
+    const consentLogsURL = isProd
+        ? 'https://consent-logs.guardianapis.com/report'
+        : 'https://consent-logs.code.dev-guardianapis.com/report';
+
+    const consentData = new ConsentString();
+    consentData.setGlobalVendorList(iabVendorList);
+    consentData.setCmpId(IAB_CMP_ID);
+    consentData.setCmpVersion(IAB_CMP_VERSION);
+    consentData.setConsentScreen(IAB_CONSENT_SCREEN);
+    consentData.setConsentLanguage(IAB_CONSENT_LANGUAGE);
+    consentData.setPurposesAllowed(allowedPurposes);
+    consentData.setVendorsAllowed(allowedVendors);
+
+    const consentStr = consentData.getConsentString();
+
+    writeIabCookie(consentStr);
+
+    const latestIabState: IabPurposeState = {
+        1: consentData.isPurposeAllowed(1),
+        2: consentData.isPurposeAllowed(2),
+        3: consentData.isPurposeAllowed(3),
+        4: consentData.isPurposeAllowed(4),
+        5: consentData.isPurposeAllowed(5),
+    };
+
+    updateStateOnSave(latestIabState);
+
+    const pAdvertising = Object.keys(latestIabState).every(
+        id => latestIabState[parseInt(id, 10)] === true,
+    );
+
+    const browserID = Cookies.get('bwid') || DUMMY_BROWSER_ID;
+
+    if (isProd && browserID === DUMMY_BROWSER_ID) {
+        throw new Error(`Error getting browserID in PROD`);
+    }
+
+    const logInfo = {
+        version: '1',
+        iab: consentStr,
+        source: 'www',
```

Comment:
This is fine for now, but this will definitely need updating before it can be rolled out more widely. It might be good to set the interface to this function now (and pass in `www` in the only case we have), rather than it be hard coded in the lib so we have to line up a change at both ends in the near future.

Up to you though, no worries at all if you're happy to go out like this and refactor later.